network/index.js

```diff
@@ -16,33 +16,29 @@
 
 import { useState, useEffect } from 'react';
 
-let unsupported;
-
-const useNetworkStatus = initialEffectiveConnectionType => {
-  if (typeof navigator !== 'undefined' && 'connection' in navigator && 'effectiveType' in navigator.connection) {
-    unsupported = false;
-  } else {
-    unsupported = true;
-  }
+const useNetworkStatus = (initialEffectiveConnectionType = null) => {
+  const supported = (typeof navigator !== 'undefined' && 'connection' in navigator && 'effectiveType' in navigator.connection)
```

Comment:
@wingleung 

Mentioned above.

Reply:
semicolon please.

Reply:
I think this initialization needs to be in the hook because we use `useState`

Reply:
hmm I get the feeling moving the supported initialization should work but it's failing some of the tests for `useNetworkStatus()` and I'm not sure if it's a jest/jsdom thing or a react hook/useState thing.

Reply:
@wingleung 

Thank you for your information.
Have you made some progress on it? I'll also try and fix it.

Reply:
had another look and fixed the network tests by using `waitForNextUpdate` to begin assertion

Reply:
nevermind, the waitForNextUpdate masked the results of the assertions. Their actually silently failing. so `supported` is still not move outside of the hook.

Reply:
I'm also wondering how moving the supported initialization outside of the hook will behave on server side rendering.

Reply:
As long as we've considered this conditional `typeof window !== 'undefined'` I think it won't throw error.

Reply:
`useMemoryStatus`, `useHardwareConcurrency`, `useHardwareConcurrency`

For above hooks or utilities, I think we have already moved `unsupported` out of the main function.

Reply:
I moved them out of the hooks for the others indeed, would be nice to move it for the network one too. The `useState` somehow needs te be treated differently in the tests than the ones without `useState`, I'll have another look at it

Reply:
@anton-karlovskiy finally got it working, the tests need to load the modules in a `jest.isolateModules()` function. This uses a sandboxed registry to ensure no states outside of the hook is being reused in the other tests.

Reply:
@wingleung 

Thank you for all your efforts through getting it work.